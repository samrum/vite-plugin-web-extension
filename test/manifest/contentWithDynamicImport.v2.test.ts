import { getResourceDir, runManifestV2Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("contentWithDynamicImport");

runManifestV2Test("contentWithDynamicImport", () => ({
  content_scripts: [
    {
      js: [`${resourceDir}/content1.js`],
      matches: ["https://*/*", "http://*/*"],
    },
    {
      js: [`${resourceDir}/content2.js`],
      matches: ["https://*/*", "http://*/*"],
    },
  ],
}));
