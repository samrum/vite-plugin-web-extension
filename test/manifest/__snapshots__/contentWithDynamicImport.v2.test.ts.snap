// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`contentWithDynamicImport - Manifest V2 1`] = `
[
  {
    "code": "function log(message) {
  console.log(message);
}
export {
  log as default
};
",
    "dynamicImports": [],
    "exports": [
      "default",
    ],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/shared/log.js",
    "fileName": "assets/log.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": true,
    "isEntry": false,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/shared/log.js": {
        "code": "function log(message) {
  console.log(message);
}",
        "originalLength": 65,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 49,
      },
    },
    "name": "log",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "(async () => {
  const log = await import(\\"../../../../log.js\\");
  log(\\"content\\");
})();
",
    "dynamicImports": [
      "assets/log.js",
    ],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/contentWithDynamicImport/content1.js",
    "fileName": "assets/test/manifest/resources/contentWithDynamicImport/content1.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/contentWithDynamicImport/content1.js": {
        "code": "(async () => {
  const log = await import('../../../../log.js');

  log(\\"content\\");
})();",
        "originalLength": 87,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 89,
      },
    },
    "name": "test/manifest/resources/contentWithDynamicImport/content1",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "(async () => {
  const log = await import(\\"../../../../log.js\\");
  log(\\"content2\\");
})();
",
    "dynamicImports": [
      "assets/log.js",
    ],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/contentWithDynamicImport/content2.js",
    "fileName": "assets/test/manifest/resources/contentWithDynamicImport/content2.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/contentWithDynamicImport/content2.js": {
        "code": "(async () => {
  const log = await import('../../../../log.js');

  log(\\"content2\\");
})();",
        "originalLength": 88,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 90,
      },
    },
    "name": "test/manifest/resources/contentWithDynamicImport/content2",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 2,
  \\"content_scripts\\": [
    {
      \\"js\\": [
        \\"test/manifest/resources/contentWithDynamicImport/content1.js\\"
      ],
      \\"matches\\": [
        \\"https://*/*\\",
        \\"http://*/*\\"
      ]
    },
    {
      \\"js\\": [
        \\"test/manifest/resources/contentWithDynamicImport/content2.js\\"
      ],
      \\"matches\\": [
        \\"https://*/*\\",
        \\"http://*/*\\"
      ]
    }
  ],
  \\"web_accessible_resources\\": [
    \\"assets/log.js\\",
    \\"assets/test/manifest/resources/contentWithDynamicImport/content1.js\\",
    \\"assets/test/manifest/resources/contentWithDynamicImport/content2.js\\"
  ]
}",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/contentWithDynamicImport/content1.js",
    "name": undefined,
    "source": "(async()=>{await import(chrome.runtime.getURL(\\"assets/test/manifest/resources/contentWithDynamicImport/content1.js\\"))})();",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/contentWithDynamicImport/content2.js",
    "name": undefined,
    "source": "(async()=>{await import(chrome.runtime.getURL(\\"assets/test/manifest/resources/contentWithDynamicImport/content2.js\\"))})();",
    "type": "asset",
  },
]
`;
